src/Compilers/Core/Portable/Operations/OperationWalker.cs

```diff
@@ -12,14 +12,11 @@ public abstract class OperationWalker : OperationVisitor
     {
         private int _recursionDepth;
 
-        internal void VisitArray<T>(ImmutableArray<T> list) where T : IOperation
+        internal void VisitArray<T>(IEnumerable<T> list) where T : IOperation
         {
-            if (!list.IsDefault)
+            foreach (var operation in list)
```

Comment:
❗️ This is heavily-allocating on a fast path. `IOperationWithChildren.Children` should not be enumerated in Roslyn code or standard analyzers in its current form.